Cache/Requests.lean

```diff
@@ -8,7 +8,7 @@ import Cache.Hashing
 
 namespace Cache.Requests
 
--- FRO cache is flaky so disable until we work out the kinks: https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/The.20cache.20doesn't.20work/near/411058849
+-- FRO cache is flaky so disable until we work out the kinks: https://leanprover.zulipchat.com/#narrow/channel/113488-general/topic/The.20cache.20doesn't.20work/near/411058849
```

Comment:
The old link also works for me (not logged in): we can leave this change, but are you sure it is required?

Reply:
You're right, I was convinced the old link didn't work when I opened it, but now both work

Reply:
Indeed, they both work, but it looks like the old one redirects to the new one, so let's keep the new one!